Mathlib/CategoryTheory/Monoidal/OfHasFiniteProducts.lean

```diff
@@ -39,6 +39,7 @@ open CategoryTheory.Limits
 section
 
 /-- A category with a terminal object and binary products has a natural monoidal structure. -/
+@[deprecated CartesianMonoidalCategory.ofHasFiniteProducts (since := "2025-10-19")]
```

Comment:
In the definition itself, could you use `CartesianMonoidalCategory.ofHasFiniteProducts`? (using `hasFiniteProductsOfHasBinaryAndTerminal`).

Reply:
Sure, although I had to drop the lemmas that are not true anymore